### PR TITLE
(svelte2tsx)blank possibly operator or property access near end of mustache before svelte compile

### PR DESCRIPTION
**Pull request by jasonlyu123**
Run a simple scan before sending the code to the svelte compiler. Because it's mostly a syntax error to have an operator right before the end of the mustache tag. This would allow the errored syntax to be emitted untouched. Allowing better dx. But this workaround should only apply to a syntax error. So we have to check if any of it is a valid syntax.
for example:
```sv
{a?.}
{a. }
{a ? }
{a &&}
{a+}
```
#538

### CODE REVIEW

File: packages/svelte2tsx/src/utils/htmlxparser.ts [hunk 1]
```diff
@@ -113,3 +114,57 @@ export function parseHtmlx(htmlx: string): Node {
     }
     return svelteHtmlxAst;
 }
+
+const possibleOperatorOrPropertyAccess = [
```
Comment (dummdidumm) [inline packages/svelte2tsx/src/utils/htmlxparser.ts:118]: For performance it might be better to use a `Set` here.

**Comment by dummdidumm** [comment]
This looks very nice. I'm trying to think of situations where this would break otherwise correct output.
If we want to be even more safe, we could try to do two passes: One without that scan, if that works -> go on; if it doesn't -> try again with the scan. What do you think?

**Comment by jasonlyu123** [comment]
Kind of wanted to just go for it XD.

**Comment by dummdidumm** [comment]
I think we need this to be configurable. `svelte-check` uses `svelte2tsx`, too, and in this case we want these errors to appear.

**Comment by jasonlyu123** [comment]
I agree. What about `emitOnTemplateError` in the option parameter of `svelte2tsx`

**Comment by dummdidumm** [comment]
Sounds good! It should be `false` by default. We also somehow need to enhance the tests so that you can invoke `svelte2tsx` with arbitrary options and test whether it fails or not. Maybe check if there's `test.js` in the test case folder and if yes invoke that method within instead.

**Review by dummdidumm — COMMENTED** [review]
Nice work! One more thing, then I we are good to merge I think: You put the `test.js` in the same folder as the tests for a conversion from svelte to tsx/jsx. The test utils are written such that if there's a `test.js`, the other tests will not be done anymore. So I think you need to split out the the `test.js` tests into separate folders. This would also make it more clear that these are two different tests.

**Comment by jasonlyu123** [comment]
Do you mean a folder in the test folder dedicated to the custom tests?

**Comment by dummdidumm** [comment]
I meant instead of 
```
html2jsx/edit-moustache
  | test.js
  | input.svelte
  | expected.jsx
```
Two sparate test cases
```
html2jsx/edit-moustache
  | input.svelte
  | expected.jsx
html2jsx/moustache-error
  | test.js
```
(same for the svelte2tsx test)

**Comment by jasonlyu123** [comment]
This means we'll make the default test use the new option. And the custom one uses the default option to test it fail?

**Comment by dummdidumm** [comment]
Oh I noticed just now that `test.js` actually contains two test cases, I thought it was only one and that the svelte->jsx/tsx test would be swallowed by that. Nevermind then 😄 I'll merge this then if you don't have any objections.

**Comment by jasonlyu123** [comment]
Ok! no other to adjust.

**Pull request merged**